Ginger/Ginger/Actions/_Common/ActUIElementLib/ActUIElementEditPage.xaml.cs

```diff
@@ -286,183 +466,113 @@ public Page GetConfigPage(List<ElementConfigControl> configControlsList)
-                        Width = 600,
-                        Margin = new Thickness(10, 0, 0, 0)
-                    };
-                    comboBox.Init(mAction.GetOrCreateInputParam(element.BindedString), isVENeeded: true,context:Context.GetAsContext(mAction.Context));
+                    elementLabel = CretateLabel(element);
+                    comboBox = CreateComboBox(element);
+
+                    comboBox.Init(mAction.GetOrCreateInputParam(element.BindedString), isVENeeded: true, context: Context.GetAsContext(mAction.Context));
                     ((Ginger.UserControlsLib.UCComboBox)comboBox).ComboBox.ItemsSource = element.PossibleValues;
                     if (mAction.ElementLocateBy == eLocateBy.POMElement)
                     {
                         ((Ginger.UserControlsLib.UCComboBox)comboBox).ComboBox.SelectedValue = element.DefaultValue;
                         comboBox.ComboBoxObject.Style = this.FindResource("$FlatEditInputComboBoxStyle") as Style;
-                    }                    
+                    }
                     dynamicPanel.Children.Add(elementLabel);
                     dynamicPanel.Children.Add(comboBox);
                 }
                 else if (element.ControlType == eElementType.TextBox)
                 {
-                    elementLabel = new Label()
-                    {
-                        Style = this.FindResource("$LabelStyle") as Style,
-                        Content = element.Title + ":",
-                        HorizontalAlignment = HorizontalAlignment.Left,
-                        VerticalAlignment = VerticalAlignment.Center,
-                        FontSize =14
-                    };
-                    Ginger.Actions.UCValueExpression txtBox = new Ginger.Actions.UCValueExpression()
-                    {                       
-                        Name = element.Title.ToString().Replace(" ", ""),
-                        HorizontalAlignment = HorizontalAlignment.Left,
-                        VerticalAlignment = VerticalAlignment.Center,
-                        Width = 600,
-                        Margin = new Thickness(10, 0, 0, 0)
-                    };
+                    elementLabel = CretateLabel(element);
+                    UCValueExpression txtBox = CreateTextBox(element);
 
                     txtBox.Init(Context.GetAsContext(mAction.Context), mAction.GetOrCreateInputParam(element.BindedString), isVENeeded: true);
                     ((Ginger.Actions.UCValueExpression)txtBox).ValueTextBox.Text = element.PossibleValues.ElementAt(0);
                     dynamicPanel.Children.Add(elementLabel);
                     dynamicPanel.Children.Add(txtBox);
                 }
+
+                //Added for widgets if Element Action is TriggerJavaScriptEvent
+                else if (element.ControlType == eElementType.CheckBox &&  Convert.ToBoolean(mAction.GetInputParamValue(Fields.IsWidgetsElement)))
```

Comment:
This is good but can be done in more generic way.
e.g. if need another checkbox for someother purpose i will need to add another if condition.

i think we do the event part dynamically as well and then no need to check the condition